pkg/errors/resource_reference_error.go

```diff
@@ -0,0 +1,86 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```

Comment:
nit: you can rename this file to just `pkg/errors/resource_reference.go` since the `_error` part of the filename is redundant.